### DISCUSSION TITLE
RefinementList trigger transform-items (vue instantsearch)

### DISCUSSION THREAD

**Discussion by oliverbeck-dev** [algolia/instantsearch#6101] [Q&A]
Hi,
I have a RefinementList where I already use :transform-items="myTranformFunction". this function gets called only when the widget is loaded. I need to trigger transform-items from a watcher

is this even possible?

**Comment by Haroenv** [comment]
This will only be called when InstantSearch renders. What use case do you have for re-transforming later? some async data?

**Reply by oliverbeck-dev** [reply]
exactly. Basically I have a searchable refinement list. while searching, the items (around 4500 options) come from an external datasource async.

**Reply by Haroenv** [reply]
Weird that this is not possible with Algolia out of the box 🤔 you can actually set the facet as `searchable` and then it will work out of the box.

If that's not possible for some reason (please let us know so we can list the feature gap), I recommend you either:

- fork refinementList and replace `searchForFacetValues` with your third-party request
- pass a custom `searchClient` (a la https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/) which changes the `searchForFacetValues` method of the client to call your backend instead
